include/oneapi/dpl/pstl/execution_impl.h

```diff
@@ -29,100 +30,131 @@ namespace dpl
 namespace __internal
 {
 
-/* predicate */
+//------------------------------------------------------------------------
+// backend selector with tags
+//------------------------------------------------------------------------
+
+#if _ONEDPL_PAR_BACKEND_TBB
+using __par_backend_tag = __tbb_backend_tag;
+#elif _ONEDPL_PAR_BACKEND_OPENMP
+using __par_backend_tag = __omp_backend_tag;
+#elif _ONEDPL_PAR_BACKEND_SERIAL
+using __par_backend_tag = __serial_backend_tag;
+#else
+#    error "Parallel backend was not specified"
+#endif
+
+template <class _IsVector>
+struct __serial_tag
+{
+    using __is_vector = _IsVector;
+    using __backend_tag = __par_backend_tag;
+};
 
-template <typename _Tp>
-::std::false_type __lazy_and(_Tp, ::std::false_type)
+template <class _IsVector>
+struct __parallel_tag
 {
-    return ::std::false_type{};
-}
+    using __is_vector = _IsVector;
+    using __backend_tag = __par_backend_tag;
+};
+
+struct __parallel_forward_tag
+{
+    using __is_vector = ::std::false_type;
+    using __backend_tag = __par_backend_tag;
+};
+
+template <class _IsVector, class... _IteratorTypes>
+using __tag_type = ::std::conditional_t<
+    __internal::__is_random_access_iterator_v<_IteratorTypes...>, __parallel_tag<_IsVector>,
+    ::std::conditional_t<__is_forward_iterator_v<_IteratorTypes...>, __parallel_forward_tag, __serial_tag<_IsVector>>>;
 
-template <typename _Tp>
-inline _Tp
-__lazy_and(_Tp __a, ::std::true_type)
+template <class... _IteratorTypes>
+__serial_tag<std::false_type>
+__select_backend(oneapi::dpl::execution::sequenced_policy, _IteratorTypes&&...)
```

Comment:
Looks like this and the next `__select_backend` formats is incompatible with 
```C++
template <class... _IteratorTypes, typename _KernelName>
__hetero_tag<__device_backend_tag>
__select_backend(const execution::device_policy<_KernelName>&, _IteratorTypes&&...);

#if _ONEDPL_FPGA_DEVICE
template <class... _IteratorTypes, unsigned int _Factor, typename _KernelName>
__hetero_tag<__fpga_backend_tag>
__select_backend(const execution::fpga_policy<_Factor, _KernelName>&, _IteratorTypes&&...);
#endif
```
I mean the `const` keyword.
@rarutyun  Let's add `const` for all host policies too?

Reply:
Leaved as is for compatibility with up-stream.